ts/sagas/messages.ts

```diff
@@ -72,14 +87,69 @@ export function* loadMessage(
     | undefined = yield call(getMessage, { id });
 
   if (!response || response.status !== 200) {
-    return response ? response.value : Error();
+    const error: Error = response ? response.value : Error();
+    yield put(loadMessageFailure(error));
+    return error;
   } else {
     // Trigger an action to store the new message (converted to plain object)
     yield put(loadMessageSuccess(toMessageWithContentPO(response.value)));
     return response.value;
   }
 }
 
+function* navigateToMessageDetailsSaga(
+  action: MessageDetailsNavigate
+): Iterator<Effect> {
+  // Get the SessionToken from the store
+  const sessionToken: SessionToken | undefined = yield select(
+    sessionTokenSelector
+  );
+
+  if (sessionToken) {
+    const backendClient = BackendClient(apiUrlPrefix, sessionToken);
+    const messageId = action.payload;
+
+    // tslint:disable-next-line:no-let
+    let message: MessageByIdState = yield select(
+      messageByIdSelector(messageId)
+    );
+
+    if (!message) {
+      yield call(loadMessage, backendClient.getMessage, messageId);
+      // tslint:disable-next-line:saga-yield-return-type
+      message = yield select(messageByIdSelector(messageId));
```

Comment:
this can fail indefinitely (e.g. the network is not available)

Reply:
a safe behavior could be that if `message` doesn't exist at this point, there was a network problem and we should just navigate to the messages screen and let the user issue a pull to refresh (?)

@fpersico @matteodesanti what do you think?

Reply:
@spizzichi what do you think?

Reply:
I think the better way to handle this is: if we can't load the message detail the screen must show something like "Error loading message details [RETRY]". I prefer a big button in the center of the screen to retry and not a pull to retry in this case. Example: http://emptystat.es/image/153516180095

Reply:
@fpersico yes I agree, @matteodesanti already has a design for that

Reply:
@fpersico This is kinda like a generic error message https://invis.io/K5MUO2MP3GM#/308946817_159003368_Errore_Generico
In this case is applied to payment. You can easy generalize it.
The top bar should not show any back button. The titles could be simply "error". The main blue call-to-action should allow the user to retry, the dark gray smaller one should simply take the user back to message list.

Reply:
Temporaly handled the error showing the error message with an alert.